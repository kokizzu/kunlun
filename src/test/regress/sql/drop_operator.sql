--DDL_STATEMENT_BEGIN--
CREATE OPERATOR === (
        PROCEDURE = int8eq,
        LEFTARG = bigint,
        RIGHTARG = bigint,
        COMMUTATOR = ===
);
--DDL_STATEMENT_END--

--DDL_STATEMENT_BEGIN--
CREATE OPERATOR !== (
        PROCEDURE = int8ne,
        LEFTARG = bigint,
        RIGHTARG = bigint,
        NEGATOR = ===,
        COMMUTATOR = !==
);
--DDL_STATEMENT_END--

--DDL_STATEMENT_BEGIN--
DROP OPERATOR !==(bigint, bigint);
--DDL_STATEMENT_END--

SELECT  ctid, oprcom
FROM    pg_catalog.pg_operator fk
WHERE   oprcom != 0 AND
        NOT EXISTS(SELECT 1 FROM pg_catalog.pg_operator pk WHERE pk.oid = fk.oprcom);

SELECT  ctid, oprnegate
FROM    pg_catalog.pg_operator fk
WHERE   oprnegate != 0 AND
        NOT EXISTS(SELECT 1 FROM pg_catalog.pg_operator pk WHERE pk.oid = fk.oprnegate);
		
--DDL_STATEMENT_BEGIN--
DROP OPERATOR ===(bigint, bigint);
--DDL_STATEMENT_END--

--DDL_STATEMENT_BEGIN--
CREATE OPERATOR <| (
        PROCEDURE = int8lt,
        LEFTARG = bigint,
        RIGHTARG = bigint
);
--DDL_STATEMENT_END--

--DDL_STATEMENT_BEGIN--
CREATE OPERATOR |> (
        PROCEDURE = int8gt,
        LEFTARG = bigint,
        RIGHTARG = bigint,
        NEGATOR = <|,
        COMMUTATOR = <|
);
--DDL_STATEMENT_END--

--DDL_STATEMENT_BEGIN--
DROP OPERATOR |>(bigint, bigint);
--DDL_STATEMENT_END--

SELECT  ctid, oprcom
FROM    pg_catalog.pg_operator fk
WHERE   oprcom != 0 AND
        NOT EXISTS(SELECT 1 FROM pg_catalog.pg_operator pk WHERE pk.oid = fk.oprcom);

SELECT  ctid, oprnegate
FROM    pg_catalog.pg_operator fk
WHERE   oprnegate != 0 AND
        NOT EXISTS(SELECT 1 FROM pg_catalog.pg_operator pk WHERE pk.oid = fk.oprnegate);
		
--DDL_STATEMENT_BEGIN--
DROP OPERATOR <|(bigint, bigint);
--DDL_STATEMENT_END--