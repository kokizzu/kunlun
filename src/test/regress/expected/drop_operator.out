CREATE OPERATOR === (
        PROCEDURE = int8eq,
        LEFTARG = bigint,
        RIGHTARG = bigint,
        COMMUTATOR = ===
);
CREATE OPERATOR
CREATE OPERATOR !== (
        PROCEDURE = int8ne,
        LEFTARG = bigint,
        RIGHTARG = bigint,
        NEGATOR = ===,
        COMMUTATOR = !==
);
CREATE OPERATOR
DROP OPERATOR !==(bigint, bigint);
DROP OPERATOR
SELECT  ctid, oprcom
FROM    pg_catalog.pg_operator fk
WHERE   oprcom != 0 AND
        NOT EXISTS(SELECT 1 FROM pg_catalog.pg_operator pk WHERE pk.oid = fk.oprcom);
 ctid | oprcom 
------+--------
(0 rows)

SELECT  ctid, oprnegate
FROM    pg_catalog.pg_operator fk
WHERE   oprnegate != 0 AND
        NOT EXISTS(SELECT 1 FROM pg_catalog.pg_operator pk WHERE pk.oid = fk.oprnegate);		
 ctid | oprnegate 
------+-----------
(0 rows)

DROP OPERATOR ===(bigint, bigint);
DROP OPERATOR
CREATE OPERATOR <| (
        PROCEDURE = int8lt,
        LEFTARG = bigint,
        RIGHTARG = bigint
);
CREATE OPERATOR
CREATE OPERATOR |> (
        PROCEDURE = int8gt,
        LEFTARG = bigint,
        RIGHTARG = bigint,
        NEGATOR = <|,
        COMMUTATOR = <|
);
CREATE OPERATOR
DROP OPERATOR |>(bigint, bigint);
DROP OPERATOR
SELECT  ctid, oprcom
FROM    pg_catalog.pg_operator fk
WHERE   oprcom != 0 AND
        NOT EXISTS(SELECT 1 FROM pg_catalog.pg_operator pk WHERE pk.oid = fk.oprcom);
 ctid | oprcom 
------+--------
(0 rows)

SELECT  ctid, oprnegate
FROM    pg_catalog.pg_operator fk
WHERE   oprnegate != 0 AND
        NOT EXISTS(SELECT 1 FROM pg_catalog.pg_operator pk WHERE pk.oid = fk.oprnegate);		
 ctid | oprnegate 
------+-----------
(0 rows)

DROP OPERATOR <|(bigint, bigint);
DROP OPERATOR
